packages/backend/src/apps/postman/common/throw-errors.ts

```diff
@@ -0,0 +1,33 @@
+import HttpError from '@/errors/http'
+import { generateHttpStepError } from '@/helpers/generate-step-error'
+
+type PostmanApiErrorData = {
+  code: string
+  message: string
+}
+
+export function throwSendEmailError(
+  error: HttpError,
+  position: number,
+  appName: string,
+): never {
+  let stepErrorSolution = ''
+  const postmanErrorData: PostmanApiErrorData = error.response.data
+  const errorCode = postmanErrorData.code
+  // catch common errors by users and provide solution
+  if (errorCode === 'invalid_template') {
+    stepErrorSolution =
+      'Recipient email is blacklisted. Please request for the recipient email to be whitelisted at: https://guide.postman.gov.sg/api-guide/programmatic-email-api/send-email-api/recipient-blacklist.'
+  } else if (errorCode === 'rate_limit') {
+    stepErrorSolution =
+      'Too many requests were made, please retry the failed execution after a minute.'
+  } else if (errorCode === 'service_unavailable') {
+    stepErrorSolution =
+      'Twilio service is currently unavailable, please retry the failed execution after a few minutes.'
+  } else {
+    // return original error if not caught
+    throw error
+  }
```

Comment:
nit: Looks like this could use a switch statement

Reply:
i think i will need to do nested checks here and it would look cleaner with if else because i would need to call `throw generateHttpStepError` on every switch statement, because i need to call break on it